Cargo.toml

```diff
@@ -13,6 +13,7 @@ version       = "0.4.7"
 [dependencies]
 serde      = { version = "1.0", default-features = false, features = ["alloc"] }
 serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
+snafu      = { version = "0.8.3" }
```

Comment:
I assume this choice is because of no-std support?

Reply:
snafu in general or the version specifically? 

if snafu, I just prefer it over thiserror and definitely favor it over hand-rolled errors. The major driving force for adding it though was `Backtrace`.

If the version, I didn't do an exhaustive check to see which versions are backward compat.

Reply:
Not the version, just the choice of crate. snafu is bit heavyweight, but if you deliberately chose it over thiserror, it's fine. I think hand-rolled errors would be fine too, at the scale of complexity of this crate.

Reply:
So the way `Backtrace` works is it is (pretty much) zero-cost if you do not have backtraces enabled - they are only captured if enable them via `RUST_BACKTRACE`. If a source error has a backtrace, you can omit it on your own wrapper by including `#[snafu(backtrace)]`.

The major benefit to me was simply that I could have that backtrace available on as many leaf errors as possible as I'm using `snafu` in my work-in-progress json schema crate.

Having said that, other dependencies aren't going to have it, such as`url` and `urn` so it won't be a huge loss if I go back to hand-rolled.